### ISSUE TITLE
Generic_Sniffs_WhiteSpace_DisallowSpaceIndentSniff cannot deal with alignment vs indentation

### ISSUE THREAD

**Issue by schlessera** [squizlabs/PHP_CodeSniffer#1735]
The sniff `Generic_Sniffs_WhiteSpace_DisallowSpaceIndentSniff` does not properly recognize the difference between _indentation_ (denoting the change of scope, usually done with tabs) and _alignment_ (positioning within same scope, usually done with spaces). I'm using these terms as they are visually represented [in this blog article](https://dmitryfrank.com/articles/indent_with_tabs_align_with_spaces).

Here's an example, with a multi-line conditional:
```php
<?php
function my_function() {
	if ( $first_condition === true
	     || $second_condition === true
	     || $answer !== 42 ) {
		// Do something here.
	}
}
```

The above uses 1 tab in all three lines within the function to denote that all conditions are being processed at the same level.
However, the second and third one are also preceded by 5 spaces, to align them with the beginning of the first one.

The current implementation of the sniff has the concept of "precision indentation", which I assume is meant to relate to _alignment_. However, it assumes that this precision indentation should always be smaller in number of spaces than the `$_tabWidth`.

With the default tab width of 4, this makes the code above fail, while it would not fail with a tab width of 6. So, the value of differentiating between indentation and alignment is moot, as both are directly depending on the tab width again.

Alignment with spaces (that follows after tabs and is not mixed with them) should be acceptable regardless of the number of space chars, as the needed alignment depends on the preceding code that it should be aligning to.

**Comment by schlessera** [comment]
Here's a quick screenshot that shows the tabs and spaces (which are visually indistinguishable in the above GitHub syntax highlighter):

<img width="319" alt="image 2017-10-29 at 2 33 57 pm" src="https://user-images.githubusercontent.com/83631/32144249-4a741f8e-bcb6-11e7-90e1-985ac4482d05.png">

It can be clearly seen that the tabs properly convey the "level" or "scope" of the code, whereas the spaces are only meant for _aligning_ things that should be easily read in order.

**Comment by jrfnl** [comment]
Duplicate of #1586 

Loosely related to: #1599, PR #1607 and merged PR #1404

**Comment by gsherwood** [comment]
Yep, duplicate. So closing.
